openedx/tests/xblock_integration/xblock_testcase.py

```diff
@@ -0,0 +1,202 @@
+        )
+
+        cls.chapter = ItemFactory.create(
+            parent=cls.course,
+            display_name='Overview',
+            category='chapter'
+        )
+
+        cls.urls = {}
+
+        with cls.store.bulk_operations(cls.course.id, emit_signals=False):
+            for section in cls.test_configuration:
+                section_item = ItemFactory.create(
+                    parent=cls.chapter,
+                    display_name=section['urlname'],
+                    category='sequential'
+                )
+
+                cls.urls[section['urlname']] = reverse(
+                    'courseware_section',
+                    kwargs={
+                        'course_id': unicode(cls.course.id),
+                        'chapter': 'Overview',
+                        'section': section['urlname'],
+                    }
+                )
+                print section['urlname']
+                print cls.urls[section['urlname']]
+
+                blocks = section["xblocks"]
+
+                unit_item = ItemFactory.create(
+                    parent=section_item,
+                    display_name="unit_"+section['urlname'],
+                    category='vertical'
+                )
+                for block in blocks:
+                    block_item = ItemFactory.create(
+                        parent=unit_item,
+                        category=block['blocktype'],
+                        display_name=block['urlname']
+                    )
+            cls.course_url = reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(cls.course.id),
+                    'chapter': 'Overview',
+                    'section': section['urlname'],
+                }
+            )
+            print cls.course_url
+
+            cls.course_url = reverse(
+                'courseware_section',
+                kwargs={
+                    'course_id': unicode(cls.course.id),
+                    'chapter': u'Overview',
+                    'section': u'two_done_block_test',
+                }
+            )
+            print cls.course_url
+
+    def _containing_section(self, block_urlname):
+        '''
+        For a given block, return the parent section
+        '''
+        for section in self.test_configuration:
+            blocks = section["xblocks"]
+            for block in blocks:
+                if block['urlname'] == block_urlname:
+                    return section['urlname']
+        raise Exception("Block not found "+block_urlname)
+
+    def assertXBlockScreenshot(self, block_urlname, rendering=None):
+        '''
+        As in Bok Choi, but instead of a CSS selector, we pass a
+        block_id. We may want to be able to pass an optional selector
+        for picking a subelement of the block.
+
+        This confirms status code, and that the screenshot is
+        identical.
+
+        TODO: IMPLEMENT
+        '''
+        pass
```

Comment:
Virtually all of the XBlocks I've written have scenarios with multiple blocks. It's very easy to miss including the `element` in a CSS selector, having an id instead of a class, initializing multiple times, have overlapping global variables, or any of the other issues which can make multiple of the same XBlock fight with each other. 

With TDD and a test framework, I expect I'd do the same.